runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -0,0 +1,372 @@
+        IClientRouter newEndpointRouter = runtime.getRouter(endpoint);
+        Layout layout = (Layout) runtime.getLayoutView().getLayout().clone();
+        // Ignoring call result as the call returns ACK or throws an exception.
+        newEndpointRouter.getClient(LayoutClient.class)
+                .bootstrapLayout(layout).get();
+        newEndpointRouter.getClient(ManagementClient.class)
+                .bootstrapManagement(layout).get();
+        newEndpointRouter.getClient(ManagementClient.class)
+                .initiateFailureHandler().get();
+
+        log.info("bootstrapNewNode: New node {} bootstrapped.", endpoint);
+    }
+
+    /**
+     * Adds a new node to the existing layout.
+     *
+     * @param currentLayout        Current layout.
+     * @param endpoint             New endpoint to be added.
+     * @param isLayoutServer       is a layout server
+     * @param isSequencerServer    is a sequencer server
+     * @param isLogUnitServer      is a log unit server
+     * @param isUnresponsiveServer is an unresponsive server
+     * @param logUnitStripeIndex   stripe index to be added into if its a log unit.
+     * @throws CloneNotSupportedException if layout clone fails.
+     * @throws QuorumUnreachableException if seal or consensus cannot be achieved.
+     * @throws OutrankedException         if consensus outranked.
+     * @throws InterruptedException       if fetching global tail interrupted.
+     * @throws ExecutionException         if fetching global tail failed.
+     */
+    public void addNode(Layout currentLayout,
+                        String endpoint,
+                        boolean isLayoutServer,
+                        boolean isSequencerServer,
+                        boolean isLogUnitServer,
+                        boolean isUnresponsiveServer,
+                        int logUnitStripeIndex)
+            throws CloneNotSupportedException, QuorumUnreachableException, OutrankedException,
+            InterruptedException, ExecutionException {
+
+        currentLayout.setRuntime(runtime);
+        sealEpoch(currentLayout);
+
+        LayoutBuilder layoutBuilder = new LayoutBuilder(currentLayout);
+        if (isLayoutServer) {
+            layoutBuilder.addLayoutServer(endpoint);
+        }
+        if (isSequencerServer) {
+            layoutBuilder.addSequencerServer(endpoint);
+        }
+        if (isLogUnitServer) {
+            layoutBuilder.addLogunitServer(logUnitStripeIndex, getMaxGlobalTail(currentLayout),
+                    endpoint);
+        }
+        if (isUnresponsiveServer) {
+            layoutBuilder.addUnresponsiveServers(Collections.singleton(endpoint));
+        }
+        Layout newLayout = layoutBuilder.build();
+        newLayout.setRuntime(runtime);
+
+        attemptConsensus(newLayout);
+
+        try {
+            // Add node is successful even if reconfigure sequencer fails.
+            // TODO: Optimize this by retrying or submitting a workflow to retry.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)